Website/src/rendering/src/components/Products/ProductImage.tsx

```diff
@@ -12,26 +12,31 @@ type ProductImageProps = {
 const ProductImage = (props: ProductImageProps): JSX.Element => {
   const [activeImg, setActiveImg] = useState(null);
 
-  const uniqueImages = [...new Map(props.images.map((image) => [image['Url'], image])).values()];
+  const uniqueImages = useMemo(
+    () => [...new Map(props.images.map((image) => [image['Url'], image])).values()],
+    [props.images]
+  );
 
-  const thumbnails = uniqueImages.length > 1 && (
-    <div className="image-secondary">
-      {uniqueImages.map((img, i) => {
-        const isActive = activeImg ? img.Url === activeImg : i === 0;
-        return (
-          <div key={img.Url} className={isActive ? 'active' : ''}>
-            <img
-              src={addTransformation(img.Url, 'w320')}
-              alt=""
-              onClick={() => setActiveImg(img.Url)}
-            />
-          </div>
-        );
-      })}
-    </div>
+  useEffect(() => setActiveImg(null), [props]);
+
+  const thumbnails = useMemo(
+    () =>
+      uniqueImages.length > 1 && (
+        <div className="image-secondary">
+          {uniqueImages.map((img, i) => {
+            const isActive = activeImg ? img.Url === activeImg : i === 0;
+            return (
+              <div key={img.Url} className={isActive ? 'active' : ''}>
+                <img src={`${img.Url}&t=w320`} alt="" onClick={() => setActiveImg(img.Url)} />
+              </div>
+            );
+          })}
+        </div>
+      ),
+    [activeImg, uniqueImages]
   );
 
-  const getActiveImage = () => {
+  const getActiveImage = useMemo(() => {
```

Comment:
Minor point, I'd suggest you either change it to useCallback (and verify it still works correctly) or to another variable name e.g. displayedImage instead of getActiveImage just to make sure the naming is clear.